core/indexer/elasticsearch.go

```diff
@@ -347,28 +348,50 @@ func (ei *elasticIndexer) buildTransactionBulks(
 			}
 
 			currentTx, ok := currentTxHandler.(*transaction.Transaction)
-			if !ok {
-				ei.logger.Warn("elasticsearch found tx in pool but of wrong type")
+			if ok && currentTx != nil {
+				bulks[currentBulk] = append(bulks[currentBulk], &Transaction{
```

Comment:
separate in two functions accordint to transaction type. You can create a function getTransactionType it gets input a data.transactionhandler and returns an uint. switch case according to those types.

Reply:
I've done it a bit different than suggested, because having a simple `getTransactionType` would mean I would have to do the type assertions twice (once to find the type, second in the process of building the resulted transaction since I would need access to the fields that are not on the interface).
I've implemented a getTransactionByType function that calls another two functions depending on the condition branch.